File: employee-onboarding-system-frontend/src/types/notification.ts
import type { UserRole } from './auth';

export type NotificationType =
  | 'REQUEST_CREATED'
  | 'REQUEST_APPROVED'
  | 'REQUEST_REJECTED'
  | 'REQUEST_RESUBMITTED'
  | 'FINANCE_APPROVED'
  | 'IT_COMPLETED';

export interface NotificationItem {
  id: number;
  requestId: number;
  targetRole: UserRole;
  type: NotificationType;
  title: string;
  message: string;
  read: boolean;
  createdAt: string;
}

File: employee-onboarding-system-backend/src/main/java/com/company/employee_onboarding_system/entity/Notification.java
package com.company.employee_onboarding_system.entity;

import com.company.employee_onboarding_system.enums.NotificationType;
import com.company.employee_onboarding_system.enums.Role;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long requestId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Role targetRole;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NotificationType type;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String message;

    private boolean read;

    private LocalDateTime createdAt;

    @PrePersist
    public void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.read = false;
    }
}